src/components/NodeProperties/AbstractNodeProperties/AbstractNodeProperties.tsx

```diff
@@ -0,0 +1,43 @@
+import { Stack, StackItem, TextField, PrimaryButton} from "@fluentui/react";
+import useStore from "../../../store";
+import Card from "../../Card/Card";
+import { NodePropertiesProps } from "../NodeProperties";
+import { AbstractNode } from "../../../types/polyglotElements/nodes/AbstractNode";
+import subFlow from "../../../data/abstractExample";
+import { PolyglotFlow } from "../../../types/polyglotElements";
+
+
+export type AbstractNodePropertiesProps = AbstractNode & NodePropertiesProps;
+
+
+const AbstractNodeProperties = (props: AbstractNodePropertiesProps) => {
+    const selectedNode = useStore(state => state.getSelectedNode()) as AbstractNode;
+    const updateNode = useStore(state => state.updateNode);
+
+    const { id } = selectedNode.reactFlow;
+
+    function generateSubtree () {
+        useStore.getState().removeNode(id);
+        useStore.getState().addSubFlow(subFlow.values().next().value as PolyglotFlow);
```

Comment:
in a real scenario, subFlow is not a variable you can have whenever you want because you must retrieve it from an API call. I suggest you to change the api.ts file by adding
```javascript
loadAbstractFlowElementsAsync: (flowId: string): Promise<AxiosResponse<PolyglotFlow>> => {
    const flow = subFlow.get(flowId);
    return Promise.resolve({
        data: flow!,
        status: flow ? 200 : 404,
        statusText: flow ? "OK" : "Not Found",
        headers: {},
        config: {},
    });
},
```
to the API object.

When you need to retrieve the subflow, you should use a function similar to the following one [(here you can see how it is used)](https://github.com/polyglot-edu/node-editor/blob/cd237c7881b24fd7e651837d2c363cd1fd1bd2c4/src/components/AppFlowProvider/AppFlowProvider.tsx#L21) which also handles errors, flow not found, and sets the component state accordingly.

```javascript
async () => {
    setLoading.setTrue();
    setError(null);
    try {
        const flowElements = await fetchFunction(flowId ?? "");
        if (flowElements.status === 200) {
            console.log("flow elements loaded 🆗");
            setLoading.setFalse();
            useStore.getState().loadFlow(flowElements.data);
        } else {
            console.error("flow elements not loaded 😢");
            setError("Error loading flow elements");
            if (flowElements.status === 404) {
                setError("Flow not found");
            }
        }
    } catch (error) {
        console.error(error);
        setError("Error loading flow elements");
  
        if ((error as AxiosError).response?.status === 404) {
            setError("Flow not found");
        }
    }
    setLoading.setFalse();
}
```